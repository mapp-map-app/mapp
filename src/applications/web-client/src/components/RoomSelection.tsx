import styled from 'styled-components';
import useSWR from 'swr';
import { swrFetch } from '../utils/fetch-helpers';
import { Room } from 'mapp-types';
import SmallRoomDisplay from './SmallRoomDisplay';
import CreateRoomButton from './CreateRoomButton';

const List = styled.div`
  display: flex;
  flex-direction: column;
`;

const ListItem = styled.div`
  margin: 5px;
`;

const RoomSelection = () => {
  const { data, error } = useSWR<Room[]>('/rooms', swrFetch, {
    refreshInterval: 5000,
  });
  if (error) {
    return <p>{'Unable to load room list'}</p>;
  }

  if (!data) {
    return <List>{'Loading rooms...'}</List>;
  }

  return (
    <List>
      <ListItem>
        <CreateRoomButton />
      </ListItem>
      {data.map((room) => (
        <ListItem key={room.id}>
          <SmallRoomDisplay room={room} />
        </ListItem>
      ))}
    </List>
  );
};

export default RoomSelection;
