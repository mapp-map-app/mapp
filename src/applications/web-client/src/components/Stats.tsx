import styled from 'styled-components';
import useSWR from 'swr';
import { swrFetch } from '../utils/fetch-helpers';

const List = styled.div`
  display: flex;
`;

const ListItem = styled.div`
  margin: 5px;
`;

const Stats = () => {
  const { data, error } = useSWR<{
    connections: number;
    rooms: number;
    objects: number;
  }>('/stats', swrFetch, {
    refreshInterval: 2000,
  });
  if (error) {
    return <p>{'Unable to load stats'}</p>;
  }

  if (!data) {
    return <List>{'Loading...'}</List>;
  }

  return (
    <List>
      <ListItem>{`Connections: ${data.connections}`}</ListItem>
      <ListItem>{`Rooms: ${data.rooms}`}</ListItem>
      <ListItem>{`Room Objects: ${data.objects}`}</ListItem>
    </List>
  );
};

export default Stats;
