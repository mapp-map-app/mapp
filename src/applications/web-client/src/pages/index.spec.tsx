import renderer from 'react-test-renderer';
import Chance from 'chance';

import Index from './index';

import Stats from '../components/Stats';
import RoomSelection from '../components/RoomSelection';

jest.mock('../components/Stats');
jest.mock('../components/RoomSelection');

describe('Index', () => {
  const chance = new Chance();
  let randomUserAgent: string;
  beforeEach(() => {
    const mockStats = Stats as jest.Mock;
    mockStats.mockReturnValue('stats component');
    const mockRoomSelection = RoomSelection as jest.Mock;
    mockRoomSelection.mockReturnValue('room component');
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<Index />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
