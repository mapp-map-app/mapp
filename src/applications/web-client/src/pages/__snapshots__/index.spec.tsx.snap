// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should match snapshot 1`] = `
Array [
  <h1>
    MAPP: Map APP
  </h1>,
  <hr />,
  "stats component",
  <hr />,
  "room component",
]
`;
